src/bech32.c

```diff
@@ -26,6 +26,11 @@
 #include <include/wally_crypto.h>
 #include "script.h"
 
+typedef enum {
+    BECH32_ENCODING_NONE,
+    BECH32_ENCODING_BECH32,
+    BECH32_ENCODING_BECH32M
+} bech32_encoding;
```

Comment:
Instead of an enum for the encoding, use a single enum for the checksum and use that to differentiate the encoding (with e.g. BECH32_CHECKSUM_UNKNOWN being 0 representing BECH32_ENCODING_NONE)